src/main/scala/au/com/agiledigital/toolform/tasks/generate/docker/DockerFormatting.scala

```diff
@@ -0,0 +1,47 @@
+package au.com.agiledigital.toolform.tasks.generate.docker
+
+import au.com.agiledigital.toolform.model.{Component, Resource, SubEdgeType}
+
+object DockerFormatting {
+  def normaliseServiceName(name: String): String =
+    name
+      .replace("/", "")
+      .replace(" ", "")
+      .replace("-", "")
+      .replace("_", "")
+      .toLowerCase
+
+  def normaliseImageName(name: String): String =
+    name
+      .replace(" ", "_")
+      .replace("-", "_")
+      .toLowerCase
+
+  def componentServiceName(component: Component): String =
+    normaliseServiceName(s"${component.id}")
+
+  def resourceServiceName(resource: Resource): String =
+    normaliseServiceName(s"${resource.id}")
+
+  def subEdgeServiceName(projectId: String, subEdgeDef: SubEdgeDef): String =
+    normaliseServiceName(s"${projectId}${subEdgeDef.edgeId}${subEdgeDef.subEdgeId}nginx")
+
+  def componentImageName(projectId: String, component: Component): String =
+    normaliseImageName(s"${projectId}/${component.id}")
+
+  def resourceImageName(resource: Resource): String =
+    normaliseImageName(s"${resource.image}")
```

Comment:
Why is this using `s"..."` instead of just `resource.image`?

Reply:
Good point. All examples of this have been removed.